src/azure-cli/azure/cli/command_modules/monitor/operations/autoscale_settings.py

```diff
@@ -338,6 +338,10 @@ def autoscale_rule_create(cmd, client, autoscale_name, resource_group_name, cond
 
 def autoscale_rule_list(cmd, client, autoscale_name, resource_group_name, profile_name=DEFAULT_PROFILE_NAME):
     autoscale_settings = client.get(resource_group_name, autoscale_name)
+    profile_names = [x.name for x in autoscale_settings.profiles]
+    if not profile_name in profile_names:
+        from knack.util import CLIError
+        raise CLIError('Profile name is invalid. Please check the existence of the profile.')
     profile = next(x for x in autoscale_settings.profiles if x.name == profile_name)
```

Comment:
Or, could use try...except sinppet to catch StopIteration error, then raise CLIError.

Reply:
well. So `next` cannot handle such scenario and we need extra logic. Right? Compared to try..except, I prefer current one.

Reply:
using try..except is the same what you did now.